recordCollection.go

```diff
@@ -0,0 +1,196 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/miekg/dns"
+)
+
+type (
+	recordCollection []cloudflare.DNSRecord
+)
+
+// Find will serach for needle in a recordCollection.
+func (c recordCollection) Find(needle cloudflare.DNSRecord) (int, *cloudflare.DNSRecord) {
+	for i, r := range c {
+		if match(r, needle) {
+			return i, &r
+		}
+	}
+
+	return -1, nil
+}
+
+// Diff will find the differences between two recordCollections.
+func (c recordCollection) Diff(remote recordCollection) (recordCollection, recordCollection) {
+	localOnly := recordCollection{}
+	remoteOnly := recordCollection{}
+
+	for _, l := range c {
+		n, _ := remote.Find(l)
+
+		if n < 0 {
+			localOnly = append(localOnly, l)
+		}
+	}
+
+	for _, r := range remote {
+		n, _ := c.Find(r)
+
+		if n < 0 {
+			remoteOnly = append(remoteOnly, r)
+		}
+	}
+
+	return localOnly, remoteOnly
+}
+
+func (c recordCollection) Print(w io.Writer) {
+	maxName := 0
+	for _, r := range c {
+		if len(r.Name) > maxName {
+			maxName = len(r.Name)
+		}
+	}
+
+	for _, r := range c {
+		name := r.Name + "." + strings.Repeat(" ", maxName-len(r.Name))
+
+		proxied := ""
+		if r.Proxied {
+			proxied = " ; PROXIED"
+		}
+
+		fmt.Fprintf(w, "%s %d %-8s %s%s\n", name, r.TTL, "IN "+r.Type, r.Content, proxied)
+	}
+}
+
+// parseZone will parse a BIND style zone file and return the zone name and
+// a recordCollection.
+func parseZone(r io.Reader) (string, recordCollection, error) {
+	var zoneName string
+	records := recordCollection{}
+
+	for t := range dns.ParseZone(r, "", "") {
```

Comment:
`t` for what? `t` for "record"?
I see that the terminology comes straight out the used dns package. I still don't like it.

Reply:
Well. Let's keep in in the style of miekg/dns even if we don't like it ;)

Reply:
Ok